### ISSUE TITLE
triangulate cannot triangulate a 3 vertex polygon

### ISSUE THREAD

**Issue by cwgable** [lanl/LaGriT#56]
If an input polygon has only 3 points, triangulate fails.

**Comment by millerta** [comment]
From Carl:
I think we deal with this in the Python side of dfnWorks so it is not an issue. Even if it is not ‘fixed’ a few lines of code to detect at exit gracefully or even detect an make a 1 tringle MO would be straight forward. However, it might take 4-8 hours to do it right so since dfnWorks does not need it I would make it low priority.

**Comment by daniellivingston** [comment]
Fixed in commit 8dbfc03b59a7927b3b657b95807d6efe1e385d79. Please review @millerta .

Modified `src/triangulate_lg.f` and added test in `test/level01/single_triangle/`.

Previously, triangulating and dumping a three-node mesh generates this AVS file:

```
          3          1          4          0          0
 001   0.000000000000E+00  0.000000000000E+00  0.000000000000E+00
 002   1.000000000000E+00  0.000000000000E+00  0.000000000000E+00
 003   5.000000000000E-01  7.500000000000E-01  0.000000000000E+00
 001        0   tri ** ** **
 00004  1  1  1  1
 imt1, integer
 itp1, integer
 icr1, integer
 isn1, integer
    1  0  0  0  0
    2  0  0  0  0
    3  0  0  0  0
```

With the modifications, the subroutine correctly outputs:

```
         3          1          4          0          0
001   0.000000000000E+00  0.000000000000E+00  0.000000000000E+00
002   1.000000000000E+00  0.000000000000E+00  0.000000000000E+00
003   5.000000000000E-01  7.500000000000E-01  0.000000000000E+00
001        0   tri  3  1  2
00004  1  1  1  1
imt1, integer
itp1, integer
icr1, integer
isn1, integer
   1  0  10  0  0
   2  0  10  0  0
   3  0  10  0  0
```
